python/taichi/lang/transformer.py

```diff
@@ -423,21 +344,99 @@ def is_decorated(iter):
-        else:  # Struct for
-            assert is_static_for == 0
-            assert is_ndrange_for == 0
-            if isinstance(node.target, ast.Name):
-                elts = [node.target]
+
+
+        def visit_ndrange_for(node):
+            # for i, j in ti.ndrange(n)
+            template = '''
+if ti.static(1):
+    __ndrange = 0
+    for __ndrange_I in range(0):
+        __I = __ndrange_I
+            '''
+            t = ast.parse(template).body[0]
+            t.body[0].value = node.iter
+            t_loop = t.body[1]
+            t_loop.iter.args[0] = self.parse_expr(
+                '__ndrange.acc_dimensions[0]')
+            targets = get_targets()
+            targets_tmp = ['__' + name for name in targets]
+            loop_body = t_loop.body
+            for i in range(len(targets)):
+                if i + 1 < len(targets):
+                    stmt = '{} = __I // __ndrange.acc_dimensions[{}]'.format(
+                        targets_tmp[i], i + 1)
+                else:
+                    stmt = '{} = __I'.format(targets_tmp[i])
+                loop_body.append(self.parse_stmt(stmt))
+                stmt = '{} = {} + __ndrange.bounds[{}][0]'.format(
+                    targets[i], targets_tmp[i], i)
+                loop_body.append(self.parse_stmt(stmt))
+                if i + 1 < len(targets):
+                    stmt = '__I = __I - {} * __ndrange.acc_dimensions[{}]'.format(
+                        targets_tmp[i], i + 1)
+                    loop_body.append(self.parse_stmt(stmt))
+            loop_body += node.body
+
+            node = ast.copy_location(t, node)
+            return self.visit(node)  # further translate as a range for
+
+
+        def visit_grouped_ndrange_for(node):
+            from astpretty import pprint
+            # for I in ti.grouped(ti.ndrange(n, m))
+            target = node.target.id
+            template = '''
+if ti.static(1):
+    __ndrange = 0
+    {} = ti.Vector([0] * len(__ndrange.dimensions))
+    for __ndrange_I in range(__ndrange.acc_dimensions[0]):
+        __I = __ndrange_I
+        for __grouped_I in range(len(__ndrange.dimensions)):
+            __grouped_I_tmp = 0
+            if __grouped_I + 1 < len(__ndrange.dimensions):
+                __grouped_I_tmp = 233
             else:
-                elts = node.target.elts
+                __grouped_I_tmp = __I
+            __tmp = 233
+            if __grouped_I + 1 < len(__ndrange.dimensions):
+                __I = 233
+            '''.format(target, target)
+            t = ast.parse(template).body[0]
+            print('ttttttttttttttttttttttttttttttttttttttttttttt')
+            pprint(t)
+            t.body[0].value = node.iter.args[0]
+            t_loop = t.body[2]
+            loop_body = t_loop.body
+            inner_loop_body = loop_body[1].body
+            inner_loop_body[1].body[0].value = self.parse_expr(
+                '__I // __ndrange.acc_dimensions[__grouped_I + 1]')
```

Comment:
I tried using `self.parse_expr` to avoid the `Expr` index in the template string but it didn't work.